pandas/core/array_algos/quantile.py

```diff
@@ -75,3 +114,50 @@ def quantile_with_mask(
         result = lib.item_from_zerodim(result)
 
     return result
+
+
+def quantile_ea_compat(
+    values: ExtensionArray, qs, interpolation: str, axis: int
```

Comment:
this is all super gross. i guess moving it here and then going to clean in future?

Reply:
depends on a few things: this can become somewhat less gross with 2D EAs (or 1D ndarrays in an ArrayManager world).  There's also the fact that values_for_factorize used here doesnt work for IntegerArray/FloatingArray.

Reply:
can this be better if we expose a .quantile() on EAs?

Reply:
that may eventually be the way to go.  will see if thats what it takes to get the IntegerArray/FloatingArray working again